packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -1267,19 +1320,10 @@ export class AnalyzerService {
         return undefined;
     }
 
-    private _getFileNamesFromFileSpecs(): Uri[] {
-        // Use a map to generate a list of unique files.
-        const fileMap = new Map<string, Uri>();
-
-        // Scan all matching files from file system.
-        timingStats.findFilesTime.timeOperation(() => {
-            const matchedFiles = this._matchFiles(this._configOptions.include, this._configOptions.exclude);
-
-            for (const file of matchedFiles) {
-                fileMap.set(file.key, file);
-            }
-        });
-
+    // Given a file map returned by the source enumerator, this function
+    // adds any open files that match the include file spec and returns a
+    // final deduped file list.
+    private _getTrackedFileList(fileMap: Map<string, Uri>): Uri[] {
```

Comment:
i wonder if we should change this to return an iterator instead.. for cases where its 100k plus entries.